### ISSUE TITLE
Crash when synchronizing in "Your Activity" screen

### ISSUE THREAD

**Issue by bparmentier** [Freeyourgadget/Gadgetbridge#662]
I already encountered three or four times a crash of the application when going to the "Your Activity" screen and tapping the _Synchronize_ button while in _Steps a week_ tab (not sure if it was always this tab).

I couldn't find a way to reproduce the bug deterministically yet.

Wearable device: MI Band 2 (FW 1.0.1.39, HW V0.1.3.2)
Gadgetbridge version: 0.18.5
Android version: CyanogenMod 14.1 (Android 7.1.1)

Here is the stack trace of the crash:

```
04-21 10:29:42.547 24811 24811 D AndroidRuntime: Shutting down VM
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: Uncaught exception: Index: 0, Size: 0java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at java.util.ArrayList.get(ArrayList.java:411) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at com.github.mikephil.charting.data.DataSet.getEntryForIndex(DataSet.java:286) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at com.github.mikephil.charting.utils.Transformer.generateTransformedValuesLine(Transformer.java:184) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at com.github.mikephil.charting.renderer.LineChartRenderer.drawValues(LineChartRenderer.java:547) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at com.github.mikephil.charting.charts.BarLineChartBase.onDraw(BarLineChartBase.java:263) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.View.draw(View.java:17189) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.View.updateDisplayListIfDirty(View.java:16171) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.View.draw(View.java:16955) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.ViewGroup.drawChild(ViewGroup.java:3727) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3513) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.View.updateDisplayListIfDirty(View.java:16166) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.View.updateDisplayListIfDirty(View.java:16134) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.View.updateDisplayListIfDirty(View.java:16134) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.View.updateDisplayListIfDirty(View.java:16134) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.View.updateDisplayListIfDirty(View.java:16134) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.View.updateDisplayListIfDirty(View.java:16134) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.View.updateDisplayListIfDirty(View.java:16134) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.View.updateDisplayListIfDirty(View.java:16134) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.View.updateDisplayListIfDirty(View.java:16134) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.ThreadedRenderer.updateViewTreeDisplayList(ThreadedRenderer.java:648) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.ThreadedRenderer.updateRootDisplayList(ThreadedRenderer.java:654) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.ThreadedRenderer.draw(ThreadedRenderer.java:762) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.ViewRootImpl.draw(ViewRootImpl.java:2800) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:2608) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2215) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1254) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6343) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:874) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.Choreographer.doCallbacks(Choreographer.java:686) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.Choreographer.doFrame(Choreographer.java:621) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:860) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.os.Handler.handleCallback(Handler.java:751) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.os.Handler.dispatchMessage(Handler.java:95) ~[na:0.0]
04-21 10:29:42.570 24811 24811 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: 	at android.os.Looper.loop
--------- beginning of crash
04-21 10:29:42.571 24811 24811 E AndroidRuntime: FATAL EXCEPTION: main
04-21 10:29:42.571 24811 24811 E AndroidRuntime: Process: nodomain.freeyourgadget.gadgetbridge, PID: 24811
04-21 10:29:42.571 24811 24811 E AndroidRuntime: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at java.util.ArrayList.get(ArrayList.java:411)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at com.github.mikephil.charting.data.DataSet.getEntryForIndex(DataSet.java:286)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at com.github.mikephil.charting.utils.Transformer.generateTransformedValuesLine(Transformer.java:184)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at com.github.mikephil.charting.renderer.LineChartRenderer.drawValues(LineChartRenderer.java:547)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at com.github.mikephil.charting.charts.BarLineChartBase.onDraw(BarLineChartBase.java:263)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.View.draw(View.java:17189)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:16171)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.View.draw(View.java:16955)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.ViewGroup.drawChild(ViewGroup.java:3727)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3513)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:16166)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:16134)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:16134)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:16134)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:16134)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:16134)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:16134)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:16134)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:16134)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.ThreadedRenderer.updateViewTreeDisplayList(ThreadedRenderer.java:648)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.ThreadedRenderer.updateRootDisplayList(ThreadedRenderer.java:654)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.ThreadedRenderer.draw(ThreadedRenderer.java:762)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.ViewRootImpl.draw(ViewRootImpl.java:2800)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:2608)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2215)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1254)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6343)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:874)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.Choreographer.doCallbacks(Choreographer.java:686)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.Choreographer.doFrame(Choreographer.java:621)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:860)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:751)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:95)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:154)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6126)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
04-21 10:29:42.571 24811 24811 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
04-21 10:29:42.575   903  2431 W ActivityManager:   Force finishing activity nodomain.freeyourgadget.gadgetbridge/.activities.charts.ChartsActivity
```

I can provide a full logcat if more context is needed.

**Comment by cpfeiffer** [comment]
Thanks, I'll try to reproduce it. I think it's either a bug in the charting library that we use, or in the way we use it. I don't think it is specific to the Mi Band 2.

**Comment by marcospons84** [comment]
Same bug here, but using amazfit bip, only reproducible once synchronized first time and having some data.

**Comment by cpfeiffer** [comment]
See https://github.com/PhilJay/MPAndroidChart/pull/2958

**Comment by cpfeiffer** [comment]
Opened #917 for this.

**Comment by cpfeiffer** [comment]
The crash should be fixed by using a custom/fixed version of MPAndroidchart (as in #917)
